.pipelines/cni/lsg/kernel-upgrade-template.yaml

```diff
@@ -0,0 +1,307 @@
+  name: ""
+  clusterType: ""
+  clusterName: ""
+  nodeCount: ""
+  vmSize: ""
+  os: linux
+  arch: ""
+  osSKU: Ubuntu
+
+# CNIv2
+# + Should be able to scale up/down the pods successfully certain number of times.
+# + Node reboot scenarios should be covered.
+# + The CNS state should be validated with that of CNI state.
+# + Pods should have ips assigned and connectivity/datapath test should be present.
+# + CNS restart and validates the state
+
+stages:
+  - stage: create_${{ parameters.name }}
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+    dependsOn:
+      - setup
+    displayName: "Create Cluster - ${{ parameters.clusterName }}"
+    jobs:
+      - job: create_aks_cluster_with_${{ parameters.name }}
+        steps:
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                make -C ./hack/aks ${{ parameters.clusterType }} \
+                AZCLI=az REGION=$(LOCATION) SUB=$(SUB_AZURE_NETWORK_AGENT_BUILD_VALIDATIONS) \
+                CLUSTER=${{ parameters.clusterName }}-$(commitID) \
+                VM_SIZE=${{ parameters.vmSize }} \
+                AUTOUPGRADE=none
+            name: "CreateAksCluster"
+            displayName: "Create AKS Cluster"
+
+  - stage: ${{ parameters.name }}
+    variables:
+      commitID: $[ stagedependencies.setup.env.outputs['SetEnvVars.commitID'] ]
+      cnsVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.cnsVersion'] ]
+      cniVersion: $[ stagedependencies.setup.env.outputs['SetEnvVars.cniVersion'] ]
+    pool:
+      name: $(BUILD_POOL_NAME_DEFAULT)
+    dependsOn:
+      - create_${{ parameters.name }}
+      - setup
+    displayName: "CNIv2 Test - ${{ parameters.name }}"
+    jobs:
+      - job: integration
+        displayName: "Integration Test - ${{ parameters.name }}"
+        steps:
+          - task: AzureCLI@1
+            inputs:
+              azureSubscription: $(BUILD_VALIDATIONS_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                echo cns version - $(cnsVersion)
+                echo cni version - $(cniVersion)
+
+                echo "Start Integration Tests on Overlay Cluster"
+                make -C ./hack/aks set-kubeconf AZCLI=az CLUSTER=${{ parameters.clusterName }}-$(commitID)
+                kubectl cluster-info
+                kubectl get po -owide -A
+                sudo -E env "PATH=$PATH" make test-integration CNS_VERSION=$(cnsVersion) CNI_VERSION=$(cniVersion) INSTALL_CNS=true INSTALL_AZURE_CNI_OVERLAY=true CNS_IMAGE_REPO=$(CNS_IMAGE_REPO) CNI_IMAGE_REPO=$(CNI_IMAGE_REPO)
```

Comment:
This would be make `test-load` right ?

Reply:
Hmmm... That is a decent point. We are not testing CNI, but the integration between a kernel upgrade and an already released CNI. Ideally, I think we should just have CNS installed and proceed with the testing.